### ISSUE TITLE
Configuration and making IDE layer

### ISSUE THREAD

**Issue by MohsenNz** [neovimhaskell/nvim-hs#116]
Hi guys. Can I configure my neovim with nvim-hs and also create an IDE layer (like lunarVim) with that ?
And is it good idea at all ?

**Comment by saep** [comment]
It's definitely possible. It might also be fun. However, since most of the work is going to be integrating existing plugins and changing their configuration, it's easier and faster to do that in Lua. Haskell's type system is not going to help you there. If you were mostly using plugins written in Haskell for your IDE layer, then it would make sense to do so.

nvim-hs' strength is in creating complex plugins or plugins that use existing Haskell libraries to do their work.

It's really hard and a lot of work to provide a good vim distribution. I can only speak of experience when I've used Emacs. A distribution is rarely good enough and you sometimes need a specific plugin/package and have to add it. spacemacs has made that very complicated since I had to learn how spacemacs works to add a simple package and I hated it for it. It was also incredibly sluggish. Doom Emacs was really nice though. The extra complexity for adding other plugins isn't much. You mainly use a macro that is almost identical to vanilla Emacs packaging and add the source repository/package-name in some file. Doom also had an additional purpose, it has made loading plugins and starting Emacs noticeably faster and provided nice and extensible defaults.
